// @tsparticles/angular/esm2022/lib/ng-particles.module.mjs
import { NgModule } from '@angular/core';
import { NgxParticlesComponent } from './ng-particles.component';
import { NgParticlesService } from './ng-particles.service';
import * as i0 from "@angular/core";
export class NgxParticlesModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.8", ngImport: i0, type: NgxParticlesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.8", ngImport: i0, type: NgxParticlesModule, declarations: [NgxParticlesComponent], exports: [NgxParticlesComponent] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.8", ngImport: i0, type: NgxParticlesModule, providers: [NgParticlesService] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.8", ngImport: i0, type: NgxParticlesModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgxParticlesComponent],
                    exports: [NgxParticlesComponent],
                    providers: [NgParticlesService],
                }]
        }] });
export { NgParticlesService };
