// @tsparticles/angular/esm2022/lib/ng-particles.component.mjs
import { Component, EventEmitter, Inject, Input, Output, PLATFORM_ID, } from '@angular/core';
import { isPlatformServer } from '@angular/common';
import { from, mergeMap, Subject, takeUntil } from 'rxjs';
import { tsParticles } from '@tsparticles/engine';
import * as i0 from "@angular/core";
import * as i1 from "./ng-particles.service";
export class NgxParticlesComponent {
    platformId;
    particlesService;
    options;
    url;
    id;
    particlesInit;
    particlesLoaded = new EventEmitter();
    subscription;
    destroy$ = new Subject();
    container;
    constructor(platformId, particlesService) {
        this.platformId = platformId;
        this.particlesService = particlesService;
        this.id = 'tsparticles';
    }
    ngOnInit() {
        this.subscription = this.particlesService.getInstallationStatus().subscribe(() => {
            this.container?.destroy();
            this.loadParticles();
        });
    }
    ngAfterViewInit() {
        if (isPlatformServer(this.platformId)) {
            return;
        }
        this.loadParticles();
    }
    ngOnDestroy() {
        this.container?.destroy();
        this.subscription?.unsubscribe();
        this.destroy$.next();
    }
    loadParticles() {
        const cb = (container) => {
            this.container = container;
            this.particlesLoaded.emit(container);
        };
        from(this.particlesInit ? this.particlesInit(tsParticles) : Promise.resolve())
            .pipe(mergeMap(() => {
            return tsParticles.load({ id: this.id, url: this.url, options: this.options });
        }), takeUntil(this.destroy$))
            .subscribe(cb);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.8", ngImport: i0, type: NgxParticlesComponent, deps: [{ token: PLATFORM_ID }, { token: i1.NgParticlesService }], target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.8", type: NgxParticlesComponent, selector: "ngx-particles", inputs: { options: "options", url: "url", id: "id", particlesInit: "particlesInit" }, outputs: { particlesLoaded: "particlesLoaded" }, ngImport: i0, template: '<div [id]="id"></div>', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.8", ngImport: i0, type: NgxParticlesComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'ngx-particles',
                    template: '<div [id]="id"></div>',
                }]
        }], ctorParameters: () => [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }, { type: i1.NgParticlesService }], propDecorators: { options: [{
                type: Input
            }], url: [{
                type: Input
            }], id: [{
                type: Input
            }], particlesInit: [{
                type: Input
            }], particlesLoaded: [{
                type: Output
            }] } });
