// @tsparticles/angular/esm2022/lib/ng-particles.service.mjs
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { tsParticles } from '@tsparticles/engine';
import * as i0 from "@angular/core";
export class NgParticlesService {
    initialized = new BehaviorSubject(false);
    getInstallationStatus() {
        return this.initialized.asObservable();
    }
    async init(particlesInit) {
        await particlesInit(tsParticles);
        this.initialized.next(true);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.8", ngImport: i0, type: NgParticlesService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.8", ngImport: i0, type: NgParticlesService, providedIn: 'root' });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.8", ngImport: i0, type: NgParticlesService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
